python_modules/dagster/dagster_tests/daemon_sensor_tests/test_asset_sensor_run.py

```diff
@@ -0,0 +1,660 @@
+        \   /\  /
+          d    f
+    Sensor for d and f
+    Tests that materializing z and e does nothing, then materializing z causes d and f to materialize
+    """
+    freeze_datetime = to_timezone(
+        create_pendulum_time(year=2019, month=2, day=27, tz="UTC"),
+        "US/Central",
+    )
+    with instance_with_sensors(attribute="asset_sensor_repo") as (
+        instance,
+        workspace,
+        external_repo,
+    ):
+        with pendulum.test(freeze_datetime):
+            d_and_f_sensor = external_repo.get_external_sensor("d_and_f")
+            instance.start_sensor(d_and_f_sensor)
+
+            evaluate_sensors(instance, workspace, executor)
+
+            ticks = instance.get_ticks(
+                d_and_f_sensor.get_external_origin_id(), d_and_f_sensor.selector_id
+            )
+            assert len(ticks) == 1
+            validate_tick(
+                ticks[0],
+                d_and_f_sensor,
+                freeze_datetime,
+                TickStatus.SKIPPED,
+            )
+
+            freeze_datetime = freeze_datetime.add(seconds=60)
+        with pendulum.test(freeze_datetime):
+
+            materialize([x, e], instance=instance)
+            wait_for_all_runs_to_finish(instance)
+
+            evaluate_sensors(instance, workspace, executor)
+
+            # sensor should not start a materialize
+            ticks = instance.get_ticks(
+                d_and_f_sensor.get_external_origin_id(), d_and_f_sensor.selector_id
+            )
+            assert len(ticks) == 2
+            validate_tick(
+                ticks[0],
+                d_and_f_sensor,
+                freeze_datetime,
+                TickStatus.SKIPPED,
+            )
+
+            freeze_datetime = freeze_datetime.add(seconds=60)
+
+        with pendulum.test(freeze_datetime):
+
+            materialize([z], instance=instance)
+            wait_for_all_runs_to_finish(instance)
+
+            evaluate_sensors(instance, workspace, executor)
+
+            # sensor should materialize
+            ticks = instance.get_ticks(
+                d_and_f_sensor.get_external_origin_id(), d_and_f_sensor.selector_id
+            )
+            assert len(ticks) == 3
+            validate_tick(
+                ticks[0],
+                d_and_f_sensor,
+                freeze_datetime,
+                TickStatus.SUCCESS,
+            )
+
+            wait_for_all_runs_to_finish(instance)
+            run_request_runs = [r for r in instance.get_runs() if r.pipeline_name == "__ASSET_JOB"]
+            assert len(run_request_runs) == 1
+            assert all(
+                [r.asset_selection == {AssetKey("f"), AssetKey("d")} for r in run_request_runs]
+            )
+
+            freeze_datetime = freeze_datetime.add(seconds=60)
+
+
+@pytest.mark.parametrize("executor", get_sensor_executors())
+def test_layered_sensor(executor):
+    """Asset graph:
+        x       z       e
+        \       /\      /
+            d       f
```

Comment:
Can we also add another test where a materialization is triggered by a user for an asset that is monitored by the asset sensor? For example, with an asset selection of d, f, g where d is materialized so we can confirm that g gets materialized but d does not